integration/spark/src/main/scala/org/apache/spark/util/AlterTableUtil.scala

```diff
@@ -1105,24 +1106,33 @@ object AlterTableUtil {
         val new_column_name = newDimensionInfo
           .columnName.split(CarbonCommonConstants.POINT.toCharArray).last
         val new_column_datatype = newDimensionInfo.dataType
+
+        // check if column datatypes are altered. If altered, validate them
         if (!old_column_datatype.equalsIgnoreCase(new_column_datatype)) {
-          // datatypes of complex children cannot be altered. So throwing exception for now.
-          throw new UnsupportedOperationException(
-            "Altering datatypes of any child column is not supported")
+          this.validateColumnDataType(newDimensionInfo, oldDimensionInfo)
+          alteredDatatypesMap += (oldDimensionInfo.getColName -> newDimensionInfo)
```

Comment:
If we maintain 1 map then there would be confusion while validating columns for rename and change datatype operations. I have made a change for `alteredDatatypesMap `to store column name and `datatype ` directly instead of `newDimensionInfo `.